storage/replica.go

```diff
@@ -83,7 +83,32 @@ const (
 // returned error. Note that in a multi-replica test this filter will
 // be run once for each replica and must produce consistent results
 // each time.
-var TestingCommandFilter func(roachpb.StoreID, roachpb.Request, roachpb.Header) error
+var TestingCommandFilter TestingFilterFunc
+
+type TestingFilterFunc func(roachpb.StoreID, roachpb.Request, roachpb.Header) error
+
+func RunWithTestingCommandFilter(f func(setFilter func(TestingFilterFunc))) {
+	if TestingCommandFilter != nil {
+		panic("TestingCommandFilter not nil when attempting to set it")
+	}
+	var mu sync.RWMutex
+	var filter TestingFilterFunc
+	TestingCommandFilter = func(s roachpb.StoreID, args roachpb.Request, h roachpb.Header) error {
```

Comment:
Ah, I see; this lets you set the filter whenever you want inside. Seems good, but you can probably do a lot of this setup outside `RunWithTestingCommandFilter`, i.e. TestingCommandFilter is essentially static.

Also probably not worth having an RWMutex here.